integrations/tensorflow/bindings/python/pyiree/tf/support/tf_utils_test.py

```diff
@@ -0,0 +1,64 @@
+# Lint as: python3
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for pyiree.tf.support.tf_utils."""
+
+import os
+import tempfile
+
+from absl.testing import parameterized
+from pyiree.tf.support import tf_utils
+import tensorflow as tf
+
+
+class ConstantModule(tf.Module):
+
+  @tf.function(input_signature=[])
+  def meaning(self):
+    return tf.constant([42.])
+
+
+class UtilsTests(tf.test.TestCase, parameterized.TestCase):
+
+  @parameterized.named_parameters([
+      {
+          'testcase_name': 'single_backend',
+          'target_backends': ['vmla'],
+      },
+      {
+          'testcase_name': 'multiple_backends',
+          'target_backends': ['vmla', 'llvm'],
+      },
+  ])
+  def test_artifact_saving(self, target_backends):
+    with tempfile.TemporaryDirectory() as artifacts_dir:
+      tf_module = ConstantModule()
+      iree_compiled_module = tf_utils.compile_tf_module(
+          tf_module,
+          target_backends=target_backends,
+          artifacts_dir=artifacts_dir,
+          keep_saved_model=True)
+
+      artifacts_to_check = [
+          'saved_model',
+          f'tf_input__{"__".join(target_backends)}.mlir',
```

Comment:
So these names are now a load bearing API surface for this module. We should document them in the docstring of the compile_tf_module function.

Also, describing this in e2e/README.md would be handy too.

Reply:
SGTM

Reply:
I updated the README with a working example of compiling a `tf.Module` with `compile_tf_module` and then running one of it's functions. It's pretty ugly at the moment, but I think that should be addressed in a future change because this is too large already.